sbi/utils/conditional_density.py

```diff
@@ -321,3 +325,376 @@ def _normalize_probs(probs: Tensor, limits: Tensor) -> Tensor:
+        m: Tensor that holds the means of all mixture components.
+        mc: Tensor that holds mixture coefficients of all mixture components.
+        support: An Interval with lower and upper bounds of the support.
+    """
+
+    def __init__(self, MDN_Posterior: DirectPosterior):
+        if "MultivariateGaussianMDN" in MDN_Posterior.net.__str__():
+            # wrap copy of input object into self
+            self.__class__ = type(
+                "MDNPosterior", (self.__class__, deepcopy(MDN_Posterior).__class__), {}
+            )
+            self.__dict__ = deepcopy(MDN_Posterior).__dict__
+
+            # MoG parameters
+            self.S = None
+            self.m = None
+            self.mc = None
+            self.support = self._prior.support
+
+            self.extract_mixture_components()
+
+        else:
+            raise AttributeError("Posterior does not contain a MDN.")
+
+    @staticmethod
+    def mulnormpdf(X: Tensor, mu: Tensor, cov: Tensor) -> Tensor:
+        """Evaluates the PDF for the multivariate Guassian distribution.
+
+        Args:
+            X: torch.tensor with inputs/entries row-wise. Can also be a 1-d array if only a
+                single point is evaluated.
+            mu: torch.tensor with center/mean, 1d array.
+            cov: 2d torch.tensor with covariance matrix.
+
+        Returns:
+            prob: Probabilities for entries in `X`.
+        """
+
+        # Evaluate pdf at points or point:
+        if X.ndim == 1:
+            X = torch.atleast_2d(X)
+        sigma = torch.atleast_2d(cov)  # So we also can use it for 1-d distributions
+
+        N = mu.shape[0]
+        ex1 = torch.inverse(sigma) @ (X - mu).T
+        ex = -0.5 * (X - mu).T * ex1
+        if ex.ndim == 2:
+            ex = torch.sum(ex, axis=0)
+        K = 1 / torch.sqrt(
+            torch.pow(2 * torch.tensor(3.14159265), N) * torch.det(sigma)
+        )
+        return K * torch.exp(ex)
+
+    def check_support(self, X: Tensor) -> bool:
+        """Takes a set of points X with X.shape[0] being the number of points
+        and X.shape[1] the dimensionality of the points and checks, each point
+        for its prior support.
+
+        Args:
+            X: Contains a set of multidimensional points to check
+                against the prior support of the posterior object.
+
+        Returns:
+            within_support: Boolean array representing, whether a sample is within the
+                prior support or not.
+        """
+
+        lbound = self.support.lower_bound
+        ubound = self.support.upper_bound
+
+        within_support = torch.logical_and(lbound < X, X < ubound)
+
+        return torch.all(within_support, dim=1)
+
+    def extract_mixture_components(
+        self, x: Tensor = None
+    ) -> Tuple[Tensor, Tensor, Tensor]:
+        """Extracts the Mixture of Gaussians (MoG) parameters
+        from the MDN at either the default x or input x.
+
+        Adpated from code courtesy of @ybernaerts.
```

Comment:
Please remove. We'll acknowledge Yves in the `CHANGELOG.MD`

Reply:
Also done.